server/src/main/java/com/cloud/resourcelimit/ResourceLimitManagerImpl.java

```diff
@@ -270,7 +281,7 @@ public void incrementResourceCount(long accountId, ResourceType type, Long... de
             return;
         }
 
-        long numToIncrement = (delta.length == 0) ? 1 : delta[0].longValue();
+        long numToIncrement = (delta.length == 0) ? 1 : delta[0];
```

Comment:
`null`s may be passed as `Long`s. Not sure if this is better than calling `longValue()` on a null. I think more check is needed.

Reply:
I can revert it. No problem.

Reply:
please don´t revert. either is wrong. The original is a potention null pointer and so is this. I think
```suggestion
        long numToIncrement = (delta.length == 0) ? 1 : (delta[0] == null) ? 0 : delta[0];
```
would be guarding against all possibilities.